frontend/pages/hosts/HostDetailsPage/HostDetailsPage.jsx

```diff
@@ -230,7 +235,56 @@ export class HostDetailsPage extends Component {
     return (
       <div className="section section--packs">
         <p className="section__header">Packs</p>
-        <ul className="list">{packItems}</ul>
+        {packs.length === 0 ? (
+          <p className="info__item">No packs have this host as a target.</p>
+        ) : (
+          <ul className="list">{packItems}</ul>
+        )}
+      </div>
+    );
+  };
+
+  renderSoftware = () => {
+    const { host } = this.props;
+    const wrapperClassName = `${baseClass}__table`;
+
+    return (
+      <div className="section section--software">
+        <p className="section__header">Software</p>
+        {host.software ? (
```

Comment:
```suggestion
        {host.software.length === 0 ? (
```
Maybe this? Related to the next comment.